### PR TITLE
feat: update/delete for annoy indexer

### PR DESCRIPTION
**Pull request by cristianmtr**
Requires https://github.com/jina-ai/jina/pull/1415

### CODE REVIEW

**Review by JoanFM — COMMENTED** [review]
this is just to add a new build?

**Comment by cristianmtr** [comment]
> this is just to add a new build?

What do you mean? I added support for the masking filter, and some tests to show update / delete.

File: indexers/vector/AnnoyIndexer/__init__.py [hunk 1]
```diff
@@ -41,11 +41,11 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
         _index.build(self.n_trees)
         return _index
 
-    def query(self, keys: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.ndarray', 'np.ndarray']:
+    def query(self, query_vectors: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.ndarray', 'np.ndarray']:
```
Comment (JoanFM) [inline indexers/vector/AnnoyIndexer/__init__.py:44]: do not change the signature, I agree `keys` is not perfect, but there is a big chance that people use named arguments (we need to respect BaseClass interface)

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: indexers/vector/AnnoyIndexer/__init__.py (hunk 1 above)
Comment (cristianmtr) [reply indexers/vector/AnnoyIndexer/__init__.py]: Done

File: indexers/vector/AnnoyIndexer/tests/test_annoyindexer.py [hunk 2]
```diff
@@ -103,6 +103,32 @@ def test_annoy_indexer_known(metas):
         assert idx.shape == (4, 2)
         np.testing.assert_equal(indexer.query_by_id([7, 4]), vectors[[3, 0]])
 
+    # update
```
Comment (JoanFM) [inline indexers/vector/AnnoyIndexer/tests/test_annoyindexer.py:106]: same here

Comment (cristianmtr) [reply indexers/vector/AnnoyIndexer/tests/test_annoyindexer.py:131]: Done

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Review by florian-hoenicke — APPROVED** [review]

**Pull request merged**